IPython/frontend.py

```diff
@@ -17,7 +17,7 @@
 
 from IPython.utils.shimmodule import ShimModule, ShimWarning
 
-warn("The top-level `frontend` package has been deprecated. "
```

Comment:
Indeed, the ShimWarning was introduced in 2c9013e4, which is rel-3.0.0-881-g2c9013e , but indeed, the shim is form 1.0 !